clients/da-vinci-client/src/main/java/com/linkedin/davinci/store/AbstractStorageEngine.java

```diff
@@ -171,26 +179,30 @@ public AbstractStoragePartition getMetadataPartition() {
     return metadataPartition;
   }
 
-  public synchronized void preparePartitionForReading(int partitionId) {
-    if (!containsPartition(partitionId)) {
-      LOGGER.warn("Partition {}_{} was removed before reopening.", storeVersionName, partitionId);
-      return;
-    }
-    StoragePartitionConfig partitionConfig = new StoragePartitionConfig(storeVersionName, partitionId);
-    partitionConfig.setWriteOnlyConfig(false);
-
-    adjustStoragePartition(partitionConfig);
-  }
-
   /**
    * Adjust the opened storage partition according to the provided storagePartitionConfig.
    * It will throw exception if there is no opened storage partition for the given partition id.
+   *
+   * The reason to have {@param partitionId} is mainly used to ease the unit test.
    */
-  public synchronized void adjustStoragePartition(StoragePartitionConfig partitionConfig) {
+  public synchronized void adjustStoragePartition(
+      int partitionId,
+      StoragePartitionAdjustmentTrigger mode,
+      StoragePartitionConfig partitionConfig) {
     validateStoreName(partitionConfig);
-    int partitionId = partitionConfig.getPartitionId();
+    if (partitionId != partitionConfig.getPartitionId()) {
+      throw new VeniceException(
+          "StoragePartitionConfig should contain the right partition id: " + partitionId + ", but got "
+              + partitionConfig.getPartitionId());
+    }
+    if (!containsPartition(partitionId)) {
```

Comment:
Previously we used to throw exception for partition not found in following call to `getPartitionOrThrow`. Why do we want to silently return now?